railties/test/application/loading_test.rb

```diff
@@ -359,6 +359,18 @@ def test_initialize_can_be_called_at_any_time
     assert_predicate Rails.application, :initialized?
   end
 
+  test "frameworks aren't loaded during initialization" do
+    app_file "config/initializers/raise_when_frameworks_load.rb", <<-RUBY
+      %i(action_controller action_mailer active_job active_record).each do |framework|
```

Comment:
This list is missing two hooks:

 - `action_view` runs when the [`DebugExceptions` middleware](https://github.com/rails/rails/blob/efd3338b193da5ae5f0d42c4ad792d65e744428c/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L7) is [added to the stack](https://github.com/rails/rails/blob/efd3338b193da5ae5f0d42c4ad792d65e744428c/railties/lib/rails/application/default_middleware_stack.rb#L49); https://github.com/rails/rails/pull/28538 would fix it.
 - `action_cable` runs when [the server](https://github.com/rails/rails/blob/efd3338b193da5ae5f0d42c4ad792d65e744428c/actioncable/lib/action_cable.rb#L46) is [mounted](https://github.com/rails/rails/blob/efd3338b193da5ae5f0d42c4ad792d65e744428c/actioncable/lib/action_cable/engine.rb#L48). I'm not sure if we can avoid this one.

Ideally none of these hooks would run during initialization, but this test will prevent any more from being triggered.